api/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:16.04 AS build-env
+FROM ubuntu:16.04 AS base
```

Comment:
Can we bump this version, or will it break the Dockerfile?

Reply:
Probably, I would test it locally just to be sure

Reply:
I tried to bump it up to 18.04 but the build fails

`17 40.05 [100%] Linking CXX executable rustla2_api
#17 42.20 /usr/local/lib/libfolly.a(NestedCommandLineApp.cpp.o): In function `folly::NestedCommandLineApp::NestedCommandLineApp(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::program_options::variables_map const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)>)':
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:64: undefined reference to `boost::program_options::options_description::m_default_line_length'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:64: undefined reference to `boost::program_options::options_description::m_default_line_length'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:64: undefined reference to `boost::program_options::options_description::options_description(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, unsigned int)'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:86: undefined reference to `boost::program_options::options_description::add_options()'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:88: undefined reference to `boost::program_options::options_description_easy_init::operator()(char const*, char const*)'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:88: undefined reference to `boost::program_options::options_description_easy_init::operator()(char const*, char const*)'
#17 42.20 /usr/local/lib/libfolly.a(NestedCommandLineApp.cpp.o): In function `folly::NestedCommandLineApp::addCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::function<void (boost::program_options::variables_map const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)>)':
`

Reply:
@slugalisk

Reply:
Here's the full trace https://pastebin.com/raw/BU0j3Lgs probably something to do with the version of the compiler being different on ubuntu 18.04

Reply:
-- The C compiler identification is GNU 7.5.0                                                                                                                                               
-- The CXX compiler identification is GNU 7.5.0 
-- Boost version: 1.65.1

-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Boost version: 1.58.0